src/plugins/discover/public/utils/get_sharing_data.ts

```diff
@@ -56,10 +56,20 @@ export async function getSharingData(
 
   return {
     getSearchSource: (absoluteTime?: boolean): SerializedSearchSourceFields => {
+      const absoluteFilter = data.query.timefilter.timefilter.createFilter(index);
       const timeFilter = absoluteTime
-        ? data.query.timefilter.timefilter.createFilter(index)
+        ? absoluteFilter
         : data.query.timefilter.timefilter.createRelativeFilter(index);
 
+      // remove timeFilter from existing filter
+      if (Array.isArray(existingFilter)) {
+        existingFilter = existingFilter.filter(
+          (current) => !isEqualFilters(current, absoluteFilter)
+        );
+      } else if (isEqualFilters(existingFilter, absoluteFilter)) {
+        existingFilter = undefined;
+      }
+
```

Comment:
a first thought about this ... now is converted to absolute time when creating a filter, are we sure the generated time here matches to the existing time of the existing filter, which might also be converted, but it was converted before the creation of `absoluteFilter`. I'm not 100% sure here how this works, so I'd need to dive a bit deeper then I can this week. in any case we'd need test coverage for this.

Reply:
@kertal it does look like there could be a race condition.

This is quite tricky to reason about -- I agree tests for the different cases would be really useful (even as a way to document all this!). AFAICT there are 2 scenarios we want to handle:

1. Share data is requested for immediate use (e.g. CSV of exactly now)
2. Share data is requested for deferred use (e.g. CSV whenever + rolling window)

In case 1 we want a search source containing an absolute time range so that we can give the user an accurate picture of their data in that moment.

In case 2 we want to give users a "rolling window" so they can get their data over time (unless they've explicitly set an absolute URL, then RIP relative time filter).

A complication is in both cases there could be cascading time range filters, themselves either relative or absolute 🤪

therefore I think this code should try to "only" concern itself with the 2 scenarios and not overriding existing filters. broadly something like:

```ts
if (absoluteTime === true) {
  // we want to generate a search source that will generate a picture of the data now
  // append the absolute time filter
} else {
 // we want to generate a search source that will provide a rolling window
 // append the relative time filter
}
```

What do you guys think?

Reply:
Ah, one piece I missed is "extracting" the current time range filter in Discover because we are converting to either relative or absolute, so some overriding is required 🤔

Reply:
Alright, I think we agree then we should improve code here, but since we also need to backport I'd suggest to keep the scope small, just suggested a way we can be sure that there are no race conditions. If this is fine, it would fix the issue, and we could think about improving stuff in a future PR targeting simplification in this area. How does that sound